modules/local/gatk4/markduplicatesspark/main.nf

```diff
@@ -26,13 +28,13 @@ process GATK4_MARKDUPLICATES_SPARK {
     } else {
         avail_mem = task.memory.giga
     }
-    def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
+    prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
```

Comment:
How are you determining whether bam or cram should be the output now? Because input ending can be unequal to the output, such as bam input , cram output

Reply:
magic ;-)

More seriously, I'm using the params.skip_tools to decide that